docs/web/examples/websocket/websocket-client.py

```diff
@@ -0,0 +1,41 @@
+from typing import Any
+
+from twisted.internet.defer import Failure
+from twisted.internet.task import LoopingCall, deferLater, react
+from twisted.web.websocket import WebSocketClientEndpoint, WebSocketTransport
+
+
+class WebSocketClientDemo:
+    def textMessageReceived(self, data: str) -> None:
+        print(f"received text: {data!r}")
+
+    def negotiationStarted(self, transport: WebSocketTransport) -> None:
+        ...
+
+    def negotiationFinished(self) -> None:
+        ...
+
+    def bytesMessageReceived(self, data: bytes) -> None:
+        ...
+
+    def connectionLost(self, reason: Failure) -> None:
+        ...
+
+    def pongReceived(self, payload: bytes) -> None:
+        ...
+
+
+class MyClient:
+    def buildProtocol(self, uri: str) -> WebSocketClientDemo:
+        return WebSocketClientDemo()
+
+
+async def main(reactor: Any) -> None:
+    endpoint = WebSocketClientEndpoint.new(reactor, "ws://localhost:8080/webskt.rpy")
+    print("connecting...")
+    await endpoint.connect(MyClient())
```

Comment:
I was expecting for a simple API that reduces the boilerplace code

```suggestion
    from twisted.internet.endpoints import connectProtocol
    await connectProtocol(endpoint, WebSocketClientDemo())
```

Reply:
It's not a standard stream protocol, because we can handle both text and bytes messages, as well as framed pings and pongs. The differences are subtle, but running a bytes-based TCP-style stream protocol *over* a websocket like this is explicitly left for a future addition.